### PR TITLE
gcp - add pubsub initial setup

### PR DESCRIPTION
**Pull request by egor-miasnikov**
Added initial classes with `query` and `get` functionality for `Subscription` and `Snapshots` resources in `PubSub` service

### CODE REVIEW

File: tools/c7n_gcp/tests/test_pubsub.py [hunk 1]
```diff
@@ -0,0 +1,107 @@
...
+        project_id = 'cloud-custodian'
+        session_factory = self.replay_flight_data(
+            'pubsub-topic-query', project_id=project_id)
+
+        policy = self.load_policy(
+            self.dryrun_policy,
+            session_factory=session_factory)
+
+        pubsub_topic_resources = policy.run()
+        self.assertEqual(len(pubsub_topic_resources), 1)
```
Comment (bclements) [inline tools/c7n_gcp/tests/test_pubsub.py]: We need to be more explicit about the results we expect to be returned. What I mean by this is if we expect 7 results we should assertEqual on the specific list of 7 results. 

So if you expect a policy run to return 7 results with the names "location1, location2, location3, location4, location5, location6, location7", then verify that is what you get back in your test.

**Comment by alexkarpitski** [comment]
@egor-miasnikov @bclements the suggested changes (moved the shared test policy to methods according to other PRs) have been done.

File: tools/c7n_gcp/tests/test_pubsub.py [hunk 2]
```diff
@@ -0,0 +1,97 @@
...
+            {'project_id': project_id, 'topic_id': pubsub_topic_name})
+        self.assertEqual(pubsub_topic_resource['name'], pubsub_topic_name)
+
+
+class PubSubSubscriptionTest(BaseTest):
+
+    def test_pubsub_subscription_query(self):
+        project_id = 'cloud-custodian'
+        session_factory = self.replay_flight_data(
+            'pubsub-subscription-query', project_id=project_id)
```
Comment (jondavidi) [inline tools/c7n_gcp/tests/test_pubsub.py]: I think you might be able to use the same flight data file you have for both tests when it's a `query` and `get` test

Comment (alexkarpitski) [reply tools/c7n_gcp/tests/test_pubsub.py]: Thank you for your proposal, however, there are different API calls involved: the one for `query` (list) requires no parameters while the `get` call needs a project to be specified. To be more precise, even if the flight data were named the same way, the flight recorder adds all the parameters used to the file names which does not allow to use the same file in this case.

**Pull request closed without merging**